pkg/synthesis/synthesis_test.go

```diff
@@ -102,208 +120,249 @@ var groupsByVmsTests = []synthesisTest{
 	}
-	expectedOutputFileName := filepath.Join(getTestsDirOut(), synTest.name+suffix+".txt")
-	compareOrRegenerateOutputPerTest(t, mode, actualOutput, expectedOutputFileName, synTest.name)
 }
-
 func TestPreprocessing(t *testing.T) {
-	logging.Init(logging.LowVerbosity)
-	for i := range allTests {
-		test := &allTests[i]
-		// to generate output comment the following line and uncomment the one after
-		test.runPreprocessing(t, OutputComparison)
-		//nolint:gocritic // uncomment for generating output
-		//test.runPreprocessing(t, OutputGeneration)
+	parallelTestsRun(t, runPreprocessing)
+}
+func TestConvertToAbsract(t *testing.T) {
+	parallelTestsRun(t, runConvertToAbstract)
+}
+func TestK8SSynthesis(t *testing.T) {
+	parallelTestsRun(t, runK8SSynthesis)
+}
+func TestCompareNSXConnectivity(t *testing.T) {
+	parallelTestsRun(t, runCompareNSXConnectivity)
+}
+
+// the TestLiveNSXServer() collect the resource from live nsx server, and call serialTestsRun()
+func TestLiveNSXServer(t *testing.T) {
+	logging.Init(logging.HighVerbosity)
+	server, err := collector.GetNSXServerDate("", "", "")
+	if err != nil {
+		fmt.Println(err.Error())
+		return
 	}
+	rc, err := collector.CollectResources(server)
+	require.Nil(t, err)
+	require.NotNil(t, rc)
+	serialTestsRun(&liveNsxTest, t, rc)
 }
 
-func (synTest *synthesisTest) runConvertToAbstract(t *testing.T, mode testMode) {
-	rc := data.ExamplesGeneration(&synTest.exData.FromNSX)
-	hintsParm := &symbolicexpr.Hints{GroupsDisjoint: [][]string{}}
-	suffix := "_ConvertToAbstractNoHint.txt"
-	if !synTest.noHint {
-		hintsParm.GroupsDisjoint = synTest.exData.DisjointGroupsTags
-		suffix = "_ConvertToAbstract.txt"
+// the TestLiveNSXServer() get the resource from resources.json, and call serialTestsRun()
+func TestNsxResourceFile(t *testing.T) {
+	inputFile := filepath.Join(getTestsDirIn(), "resources.json")
+	b, err := os.ReadFile(inputFile)
+	require.Nil(t, err)
+	rc, err := collector.FromJSONString(b)
+	require.Nil(t, err)
+	require.NotNil(t, rc)
+	if len(rc.DomainList) == 0 {
+		fmt.Printf("%s has no domains\n", inputFile)
+		return
 	}
-	if synTest.allowOnlyFromCategory > 0 {
-		suffix = fmt.Sprintf("%v_%s", suffix, synTest.allowOnlyFromCategory)
+	serialTestsRun(&resourceFileTest, t, rc)
+}
+
+// ///////////////////////////////////////////////////////////////////////////////////////////////////////
+// serialTestsRun() gets a resource, and run the test functions serially
+// we need it to be serially, because we have only one resource
+func serialTestsRun(synTest *synthesisTest, t *testing.T, rc *collector.ResourcesContainerModel) {
+	runPreprocessing(synTest, t, rc)
+	runConvertToAbstract(synTest, t, rc)
+	runK8SSynthesis(synTest, t, rc)
+	runCompareNSXConnectivity(synTest, t, rc)
+}
+
+// parallelTestsRun() gets a test function to run, and run it on all the syntheticTests in parallel
+func parallelTestsRun(t *testing.T, f func(synTest *synthesisTest, t *testing.T, rc *collector.ResourcesContainerModel)) {
```

Comment:
shouldn't use ` t.Parallel() ` to run in parallel?

Reply:
fixed